### DISCUSSION TITLE
Invalid hook call and Cannot read properties of null (reading 'useRef')

### DISCUSSION THREAD

**Discussion by marcodromero** [pmndrs/zustand#1979] [Q&A] [answered]
Hello! I have a problem with the most basic of zustand. Throws errors when trying to call useOperationStore() and destruct. But if it works useOperationStore.getState(). I have tried copying the examples from the documentation and I have the same problem. The dependencies of my project are these "dependencies": {
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-router-dom": "6.3.0",
    "react-scripts": "5.0.1",
    "web-vitals": "2.1.4"
  }, and the development "devDependencies": {
    "@babel/plugin-proposal-private-property-in-object": "^7.21.11"
  }

```
//operationStore.js
import { create } from "zustand";

const initialState = {
  operation: 10,
};

export const useOperationStore = create((set) => ({
  state: initialState,
  getOperation: (operationId) => set((state) => ({ operation: 20 })),
}));

//FormDeleteOperation.js
import { useOperationStore } from "../store/operationStore";

export const FormDeleteOperation = () => {
  const { operation, getOperation } = useOperationStore();


//Google console
Warning: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app

Uncaught TypeError: Cannot read properties of null (reading 'useRef')
    at useRef (react.development.js:1630:1)
    at useSyncExternalStoreWithSelector (with-selector.development.js:51:1)
    at useStore (index.mjs:13:1)
    at useBoundStore (index.mjs:30:1)
    at FormDeleteOperation (FormDeleteOperation.js:7:1)
    at renderWithHooks (react-dom.development.js:16305:1)
    at mountIndeterminateComponent (react-dom.development.js:20074:1)
    at beginWork (react-dom.development.js:21587:1)
    at HTMLUnknownElement.callCallback (react-dom.development.js:4164:1)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:4213:1)
```

**Comment by dbritto-dev** [comment]
@marcodromero so far everything looks good but would you mind sharing us your example using codesandbox? I would like to help

**Reply by marcodromero** [reply]
It works in codesandbox but not in my project.  https://codesandbox.io/s/jolly-moon-s3kdsv?file=/src/components/formDeleteOperation.js

**Comment by marcodromero** [comment, marked as answer]
Now that I see the dependencies here... Where did I put Zustand? :'v . Reviewing I realized that for days I have Zustand in the package.json of the Backend instead of the Frontend -_- (Bad practice: I have back and front in the same folder) . Fixed!
